pkg/cmd/pipeline/start_test.go

```diff
@@ -105,7 +150,254 @@ func Test_start_trigger(t *testing.T) {
+		create := action.(k8stest.CreateActionImpl)
+		obj := create.GetObject().(*v1alpha1.PipelineRun)
+		obj.Name = "random"
+		rFunc := k8stest.ObjectReaction(tracker)
+		_, o, err := rFunc(action)
+		return true, o, err
+	})
+
+	pipeline := Command(p)
+	got, _ := tu.ExecuteCommand(pipeline, "start", pipelineName,
+		"--last",
+		"-nns")
+
+	expected := "Pipelinerun started: random\n"
+	tu.AssertOutput(t, expected, got)
+
+	pr, err := cs.Pipeline.TektonV1alpha1().PipelineRuns(p.Namespace()).Get("random", v1.GetOptions{})
+
+	if err != nil {
+		t.Errorf("Error getting pipelineruns %s", err.Error())
+	}
+
+	for _, v := range pr.Spec.Resources {
+		if v.Name == "git-repo" {
+			tu.AssertOutput(t, "some-repo", v.ResourceRef.Name)
+		}
+	}
+
+	for _, v := range pr.Spec.Params {
+		if v.Name == "rev-param" {
+			tu.AssertOutput(t, "revision1", v.Value)
+		}
+	}
+	tu.AssertOutput(t, "test-sa", pr.Spec.ServiceAccount)
+}
+
+func Test_start_pipeline_last_merge(t *testing.T) {
+
+	pipelineName := "test-pipeline"
+
+	ps := []*v1alpha1.Pipeline{
+		tb.Pipeline(pipelineName, "ns",
+			tb.PipelineSpec(
+				tb.PipelineDeclaredResource("git-repo", "git"),
+				tb.PipelineDeclaredResource("build-image", "image"),
+				tb.PipelineParam("pipeline-param-1", tb.PipelineParamDefault("somethingdifferent-1")),
+				tb.PipelineParam("rev-param", tb.PipelineParamDefault("revision")),
+				tb.PipelineTask("unit-test-1", "unit-test-task",
+					tb.PipelineTaskInputResource("workspace", "git-repo"),
+					tb.PipelineTaskOutputResource("image-to-use", "best-image"),
+					tb.PipelineTaskOutputResource("workspace", "git-repo"),
+				),
+			), // spec
+		), // pipeline
+	}
+
+	prs := []*v1alpha1.PipelineRun{
+		tb.PipelineRun("test-pipeline-run-123", "ns",
+			tb.PipelineRunLabel("tekton.dev/pipeline", pipelineName),
+			tb.PipelineRunSpec(pipelineName,
+				tb.PipelineRunServiceAccount("test-sa"),
+				tb.PipelineRunResourceBinding("git-repo", tb.PipelineResourceBindingRef("some-repo")),
+				tb.PipelineRunResourceBinding("build-image", tb.PipelineResourceBindingRef("some-image")),
+				tb.PipelineRunParam("pipeline-param-1", "somethingmorefun"),
+				tb.PipelineRunParam("rev-param", "revision1"),
+			),
+		),
+	}
+
+	objs := []runtime.Object{ps[0], prs[0]}
+	pClient, tracker := NewPipelineClient(objs...)
+
+	cs := test.Clients{
+		Pipeline: pClient,
+		Kube:     fakekubeclientset.NewSimpleClientset(),
+	}
+
+	p := &tu.Params{Tekton: cs.Pipeline, Kube: cs.Kube}
+
+	cs.Pipeline.PrependReactor("create", "pipelineruns", func(action k8stest.Action) (bool, runtime.Object, error) {
+		create := action.(k8stest.CreateActionImpl)
+		obj := create.GetObject().(*v1alpha1.PipelineRun)
+		obj.Name = "random"
+		rFunc := k8stest.ObjectReaction(tracker)
+		_, o, err := rFunc(action)
+		return true, o, err
+	})
+
+	pipeline := Command(p)
+	got, _ := tu.ExecuteCommand(pipeline, "start", pipelineName,
+		"--last",
+		"-rgit-repo=scaffold-git",
```

Comment:
Those flags are weird. Can we use either
- the short form with `=` or a space, like `-r=git-repo=scaffold-git` or `-r git-repo=scaffold-git`
- the long form (my prefered) : `--resource=git-repo-scaffold-git` or `--resource git-repo-scaffold-git`

Reply:
OK. Changed to short form like `-r=git-repo=scaffold-git
`